### ISSUE TITLE
Disappear the pip and NEW badge from menu after the page is visited

### ISSUE THREAD

**Issue by isinsusakalli** [Shopify/polaris-react#1914]
### What problem is this PR solving?

Recently we shipped Stock Transfers feature, including a NEW badge on the Transfers navigation item and a blue pip on the Products section, its parent navigation item. Merchants found it annoying that there was no way to dismiss this indicator.

![image](https://user-images.githubusercontent.com/29760062/62384154-97fc0380-b51f-11e9-92cc-3801d9d339ac.png)
![image](https://user-images.githubusercontent.com/29760062/62384158-9cc0b780-b51f-11e9-8f8a-99b9b683f49e.png)

## Expected behavior

1. Initially you should see blue indicator next to Products.
2. If you click on Products to expand it, you should also see a NEW badge next to Transfers.
3. If you click on Transfers, the NEW badge and blue indicator should disappear.

@michaelmelanson created [this PR](https://github.com/Shopify/web/pull/14345) which adds navigation tracking so that we can dismiss this indicator after the user has navigated to the section.

## Actual behavior

Even when clicked, the Product pip nor the `New` badge disappears:
![May-29-2019 15-57-12](https://user-images.githubusercontent.com/1318/58590984-4d19f300-8233-11e9-8500-3b6b97517276.gif)

## Steps to reproduce the problem

1. Put the feature pip on the main menu and a New badge in a sub menu.
2. Click it on the pip and new badge.
3. Observe neither of them disappear.

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number:
- Browser:
- Device:
- Operating System:

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash

```

**Comment by alex-page** [comment]
Hi @isinsusakalli 👋

This issue feels related to navigation implementation in web. The Polaris navigation doesn't provide a way to add notification icons to navigation items. Do you want me to move this issue to their repository?

**Comment by michaelmelanson** [comment]
You're right that this belongs more in web. The Polaris behaviour that complicated this is that there's no way for me to put _only_ have the "NEW" badge, without the pip. Merchants found the pip to be annoying and distracting, especially if they don't really use the inventory features, because it looks like a notification and wanted it to go away.
